go/vt/vtgate/planbuilder/abstract/derived.go

```diff
@@ -37,22 +37,21 @@ func (d *Derived) TableID() semantics.TableSet {
 
 // PushPredicate implements the Operator interface
 func (d *Derived) PushPredicate(expr sqlparser.Expr, semTable *semantics.SemTable) error {
-	set := semTable.ExprDeps.Dependencies(expr)
-	infoFor, err := semTable.TableInfoFor(set)
+	tableInfo, err := semTable.TableInfoFor(semTable.ExprDeps.Dependencies(expr))
```

Comment:
this doesn't look good. we are pulling out too much data from semTable. We probably should add a method to semtable instead

Reply:
Changed via 16d65e0